drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -34,6 +39,12 @@ struct adc_smartbond_data {
 	uint8_t sequence_channel_count;
 	/* Index in buffer to store current value to */
 	uint8_t result_index;
+#if defined(CONFIG_PM_DEVICE)
+	/* Flag to prevent sleep */
+	ATOMIC_DEFINE(pm_policy_state_flag, 1);
+	/* Flag to prevent sleep */
+	uint32_t gpadc_ctrl_reg;
```

Comment:
Shouldn't we also store the value of `GPADC->GP_ADC_CTRL2_REG` which is set in `adc_smartbond_channel_setup` ?